addons/mail/static/src/core/common/suggestion_service.js

```diff
@@ -148,6 +148,14 @@ export class SuggestionService {
             partners = thread.channelMembers
                 .map((member) => member.persona)
                 .filter((persona) => persona.type === "partner");
+            if (thread.channel_type === "channel") {
+                partners = new Set([
+                    ...partners,
+                    ...Object.values(this.store.Persona.records).filter(
+                        ({ group_id }) => group_id && group_id === thread.authorizedGroupId
+                    ),
```

Comment:
It would be more efficient to make fields in `Persona` and `Store` and auto-compute insert from persona. That way the insert is efficient and we don't have to parse whole `Persona.records` every time we `make a searchPartnerSuggestions`. Basically:

```js
// store_service.js
groupIdToPersonas = new Map();

// persona_model.js
group_id = Record.attr(undefined, {
   onChange() {
       this.store.groupIdToPersonas.get(this.group_id)?.delete(this);
       if (!this.group_id) {
           return;
       }
       if (!this.store.groupIdToPersonas.has(this.group_id)) {
            this.store.groupIdToPersonas.add(this.group_id, new Set());
       }
       this.store.groupIdToPersonas.add(this.group_id, this);
   }
});

delete() {
   this.store.groupIdToPersonas.get(this.group_id)?.delete(this);
   super.delete(...arguments);
},
```

That way, `searchPartnerSuggestions` is more efficient:
```js
partners = new Set([
    ...partners,
    ...(this.store.groupIdToPersonas.get(thread.authorizedGroupId) ?? [])
]);
```
_(only need to parse collection of personas that have matching group id)_

Reply:
@alexkuhn thanks for The code, i have made changes please have a look. :smile: